test/ProductConstructionService.ScenarioTests/ScenarioTests/ScenarioTests_GitHubFlow.cs

```diff
@@ -25,8 +25,8 @@ internal class ScenarioTests_GitHubFlow : ScenarioTestBase
     [SetUp]
     public void SetUp()
     {
-        using TestParameters parameters = TestParameters.GetAsync().Result;
-        SetTestParameters(parameters);
+        _parameters = TestParameters.GetAsync().Result;
```

Comment:
I know it's technically unrelated but since you went ahead and started changing this - can't we just do this in `ScenarioTestBase` and assign it there be done with it and not have to call it in every test?

Same for `ConfigureDarcArgs`? Is there any benefit to not just doing it in the ctor and not require to call this? It probably does not change between the tests anyway.

Reply:
it looks like TestParameters was also giving us, and disposing of the test folder, so this needs to be moved elsewhere

Reply:
If we need this for all tests than just having a `[SetUp]` on the parent-most class should do the trick?

Reply:
yeah, I think it can be in the `ScenarioTestClass` as a setup and teardown

Reply:
actually the directory was mostly used for darc, except one test. So we were  creating a tmp folder for every test, installing darc in it, and then cleaning it up. I changed it so it installs darc once per test class

Reply:
We mostly have 1 test per class so this didn't change things much. Why not install darc just once per test execution?

Reply:
Why can't we run darc twice in parallel?

Reply:
Ah we can, for some reason I thought it wouldn't be possible. Think I confused it with something else.
Anyway looking at the TestParameters class, I think we can just make it a `SetUpFixture` class, and use it for all tests running on that machine

Reply:
Okay, I think this is better now. I also removed the EndToEndLogic class, and made it a part of the ScenarioTestBase. The logic is still in a separate file tho